src/integration-test/java/edu/ksu/canvas/contentMigration/ContentMigrationIT.java

```diff
@@ -0,0 +1,89 @@
+    private String user;
+    private ContentMigrationWriter writer;
+    private ContentMigrationReader reader;
+
+    @Before
+    public void setUp() throws IOException {
+        // Load from root of project
+        File config = new File("integration.properties");
+        if (!config.isFile()) {
+            throw new FileNotFoundException("Failed to find configuration: " + config.getAbsolutePath());
+        }
+        Properties properties = new Properties();
+        properties.load(new FileInputStream(config));
+
+        String url = properties.getProperty("url");
+        Objects.requireNonNull(url, "Property 'url' must be set in config file.");
+        canvasApiFactory = new CanvasApiFactory(url);
+
+        String token = properties.getProperty("token");
+        Objects.requireNonNull(token, "Property 'token' must be set in config file.");
+        nonRefreshableOauthToken = new NonRefreshableOauthToken(token);
+
+        course = properties.getProperty("course");
+        Objects.requireNonNull(course, "Property 'course' must be set in config file.");
+
+        destinationCourse = properties.getProperty("destination_course");
+        Objects.requireNonNull(destinationCourse, "Property 'destination_course' must be set in config file.");
+
+        user = properties.getProperty("user");
+        Objects.requireNonNull(user, "Property 'user' must be set in config file.");
+
+        reader = canvasApiFactory.getReader(ContentMigrationReader.class, nonRefreshableOauthToken);
+        writer = canvasApiFactory.getWriter(ContentMigrationWriter.class, nonRefreshableOauthToken);
+
+    }
+
+    @Test
+    public void testCreateContentMigration() throws IOException, InterruptedException {
+        CreateCourseContentMigrationOptions createCourseContentMigrationOptions = new CreateCourseContentMigrationOptions(destinationCourse, course, MigrationType.course_copy_importer, false);
+        Optional<ContentMigration> response = writer.createCourseContentMigration(createCourseContentMigrationOptions);
+        assertNotNull(response.get().getProgressUrl());
+        Integer migrationId = response.get().getId();
+        assertNotNull(migrationId);
+        assertEquals(user, response.get().getUserId().toString());
+        assertEquals(MigrationType.course_copy_importer.toString(), response.get().getMigrationType());
+        WorkflowState workflowState = response.get().getWorkflowState();
+        assertEquals(WorkflowState.running, workflowState);
+
+        while(WorkflowState.running.equals(workflowState) || WorkflowState.pre_processing.equals(workflowState)) {
+            response = reader.getCourseContentMigration(destinationCourse, migrationId);
+            assertEquals(migrationId, response.get().getId());
+            assertEquals(new Integer(0), response.get().getMigrationIssuesCount());
+            workflowState = response.get().getWorkflowState();
+            Thread.sleep(10000);
+        }
```

Comment:
Can you add a break here so we don't get stuck in infinite loop please.

Reply:
Handling it with a maxLoops variable if that's fine?

Reply:
Of course.